native/src/routes/search/containers/SearchModalContainer.js

```diff
@@ -83,4 +106,12 @@ const SearchModalContainer = (props: PropsType) => {
   ) : null
 }
 
-export default connect<PropsType, OwnPropsType, _, _, _, _>(mapStateToProps, mapDispatchToProps)(SearchModalContainer)
+const refresh = (refreshProps: OwnPropsType) => {
+  const { navigation } = refreshProps
+  navigation.navigate({ name: SEARCH_ROUTE })
+}
+
+export default connect<PropsType, OwnPropsType, _, _, _, _>(
+  mapStateToProps,
+  mapDispatchToProps
+)(withPayloadProvider<InnerPropsType, OwnPropsType, SearchRouteType>(refresh, () => {}, true)(SearchModalContainer))
```

Comment:
Why did you choose to add the `withPayloadProvider? For refreshing behavior or to show a loading spinner while cities are not available?

Reply:
For loading spinner while cities are not available and error handling if cities-fetch failed.

Reply:
Why do we need the cities?

Reply:
To get the cityName See FeedbackContainer (line 83ff)

  getCityName = (): string => {
    const { cities, cityCode } = this.props
    return CityModel.findCityName(cities, cityCode)
  }

Reply:
Oh... and the git blame says these are my changes... I am quite sure this is wrong and the city code should be used instead. I think I messed up while removing the feedback type here... AFAICS the city is only used to call the correct endpoint and we use the city code everywhere for that, so using the city name may even break stuff...Could you check? Sorry for that, but it probably makes things a lot easier, right?

Reply:
Yeah then we would not need withPayloadProvider here. For testumgebung it was working, but there cityCode and cityName are the same. But for other cities it is broken. I will remove it.